### PR TITLE
add context and repo filter warnings

### PR DESCRIPTION
**Pull request by unclejustin**
Closes: https://github.com/sourcegraph/sourcegraph/issues/26977

https://user-images.githubusercontent.com/1855233/170144906-dbad1384-8038-42e2-aaac-448ad715f5be.mov

## Test plan

- Go to create new insight screen
- Add a context or repo filter
- Should give error for both
- Remove context or repo filter and add plain text
- Error should be removed

## App preview:

- [Web](https://sg-web-insights-warn-incorrect-search.onrender.com)
- [Storybook](https://5f0f381c0e50750022dc6bf7-cbajntykot.chromatic.com)

Check out the [client app preview documentation](https://docs.sourcegraph.com/dev/how-to/client_pr_previews) to learn more.

### CODE REVIEW

**Comment by unclejustin** [comment]
@Joelkw review request for copy.

**Review by vovakulikov — APPROVED** [review]
@unclejustin I think for this PR and our current state this is fine, but it feels like we run too many things in search validator that we do not use in the actual query validator.

File: client/web/src/enterprise/insights/pages/insights/creation/search-insight/components/form-series-input/FormSeriesInput.tsx [hunk 1]
```diff
@@ -7,17 +7,32 @@ import { Button, Card, Input, Typography } from '@sourcegraph/wildcard'
...
+        return result
+    }
+    const { isNotContext, isNotRepo } = searchQueryValidator(value || '', true)
+
+    if (!isNotContext) {
+        return 'The `context:` filter is not currently supported.'
+    }
+
+    if (!isNotRepo) {
+        return 'Do not include the `repo:` filter; if needed `repo:` will be added automatically.'
```
Comment (Joelkw) [inline client/web/src/enterprise/insights/pages/insights/creation/search-insight/components/form-series-input/FormSeriesInput.tsx:33]:
```suggestion
        return 'Do not include a `repo:` filter; add targeted repositories above, or filter repos on the filter panel after creation'
```

File: client/web/src/enterprise/insights/pages/insights/creation/search-insight/components/form-series-input/FormSeriesInput.tsx [hunk 2]
```diff
@@ -7,17 +7,32 @@ import { Button, Card, Input, Typography } from '@sourcegraph/wildcard'
...
-const validQuery = createRequiredValidator('Query is a required field for data series.')
+const validQuery = (value: string | undefined, validity: ValidityState | null | undefined): ValidationResult => {
+    const result = createRequiredValidator('Query is a required field for data series.')(value, validity)
+    if (result) {
+        return result
+    }
+    const { isNotContext, isNotRepo } = searchQueryValidator(value || '', true)
+
+    if (!isNotContext) {
+        return 'The `context:` filter is not currently supported.'
```
Comment (Joelkw) [inline client/web/src/enterprise/insights/pages/insights/creation/search-insight/components/form-series-input/FormSeriesInput.tsx:29]:
```suggestion
        return 'The `context:` filter is not supported; instead, run over all repositories and use the `context:` on the filter panel after creation'
```

**Review by Joelkw — APPROVED** [review]
Curious to see if this wraps badly! Thanks for the review tag, goal here is to also tell the user what they can do instead, so added that. (Also, to confirm – this catches `-repo:` flags too? Do we need a separate message for that?)

**Comment by unclejustin** [comment]
@Joelkw, yes it will catch -repo as well. I don't think we need any new verbiage for it. Thanks for the tweaks!

**Pull request merged**